VirtoCommerce.Platform.Data/Assets/AssetEntryService.cs

```diff
@@ -0,0 +1,169 @@
+                    }
+
+                    if (!string.IsNullOrEmpty(criteria.Language))
+                    {
+                        query = query.Where(x => x.LanguageCode == criteria.Language);
+                    }
+
+                    if (!string.IsNullOrEmpty(criteria.Group))
+                    {
+                        query = query.Where(x => x.Group == criteria.Group);
+                    }
+
+                    if (!criteria.Tenants.IsNullOrEmpty())
+                    {
+                        var tenants = criteria.Tenants.Where(x => !string.IsNullOrEmpty(x.TenantId) && !string.IsNullOrEmpty(x.TenantType)).ToArray();
+                        if (tenants.Any())
+                        {
+                            Expression<Func<AssetEntryEntity, bool>> tenantsExp = null;
+
+                            // x => x.TenantId == tenant.TenantId && x.TenantType == tenant.TenantType
+                            var paramX = Expression.Parameter(typeof(AssetEntryEntity), "x");
+                            var tenantId = Expression.MakeMemberAccess(paramX, typeof(AssetEntryEntity).GetMember("TenantId").First());
+                            var tenantType = Expression.MakeMemberAccess(paramX, typeof(AssetEntryEntity).GetMember("TenantType").First());
+                            foreach (var tenant in tenants)
+                            {
+                                var eqTenantId = Expression.Equal(tenantId, Expression.Constant(tenant.TenantId, typeof(string)));
+                                var lambdaId = Expression.Lambda<Func<AssetEntryEntity, bool>>(eqTenantId, paramX);
+                                
+                                var eqTenantType = Expression.Equal(tenantType, Expression.Constant(tenant.TenantType, typeof(string)));
+                                var lambdaType = Expression.Lambda<Func<AssetEntryEntity, bool>>(eqTenantType, paramX);
+
+                                var body = Expression.AndAlso(lambdaId.Body, lambdaType.Body);
+                                var tenantExp = Expression.Lambda<Func<AssetEntryEntity, bool>>(body, paramX);
+                                
+                                if (tenantsExp == null)
+                                    tenantsExp = tenantExp;
+                                else
+                                {
+                                    body = Expression.OrElse(tenantsExp.Body, tenantExp.Body);
+                                    tenantsExp = Expression.Lambda<Func<AssetEntryEntity, bool>>(body, paramX);
+                                }
+                            }
+
+                            query = query.Where(tenantsExp);
+                        }
+                    }
+                    
+                    var result = new AssetEntrySearchResult
+                    {
+                        TotalCount = query.Count()
+                    };
+
+                    result.Assets = query
```

Comment:
Retrieve from DB only Ids and then load them by GetByIds methods with preserving sort order